### PR TITLE
Deprecate libpe_rules

### PR DESCRIPTION
**Pull request by kgaillot**
This moves the remaining functionality of libpe_rules to libcrmcommon, leaving libpe_rules as a deprecated library containing only deprecated APIs.

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , here's the last of the rules project

File: lib/common/xml_element.c [hunk 1]
```diff
@@ -80,15 +79,13 @@ pcmk__xe_first_child(const xmlNode *parent, const char *node_name,
...
 
-    if (node_name == NULL) {
-        node_name = "(any)";    // For logging
-    }
-    if (attr_n != NULL) {
-        crm_trace("XML child node <%s %s=%s> not found in %s",
-                  node_name, attr_n, attr_v, parent_name);
+    if (attr_n == NULL) {
+        crm_trace("%s XML has no %s child",
+                  (const char *) parent->name, pcmk__s(node_name, ""));
```
Comment (wenningerk) [inline lib/common/xml_element.c:84]: dual spaces always look as if something had gone wrong. maybe 'arbitrary-node' ...

File: lib/common/xml_element.c [hunk 2]
```diff
@@ -80,15 +79,13 @@ pcmk__xe_first_child(const xmlNode *parent, const char *node_name,
...
     } else {
-        crm_trace("XML child node <%s> not found in %s",
-                  node_name, parent_name);
+        crm_trace("%s XML has no %s child with %s='%s'",
+                  (const char *) parent->name, pcmk__s(node_name, ""),
```
Comment (wenningerk) [inline lib/common/xml_element.c:87]: same as above

File: configure.ac [hunk 3]
```diff
@@ -229,16 +229,24 @@ AC_ARG_ENABLE([hardening],
...
     [AS_HELP_STRING([--enable-systemd],
         [enable support for managing resources via systemd @<:@try@:>@])]
 )
 yes_no_try "$enable_systemd" "try"
 enable_systemd=$?
 
-dnl --enable-* options: features inherent to Pacemaker
+AC_ARG_ENABLE([deprecated-libs],
+    [AS_HELP_STRING([--enable-deprecated-libs],
+        [Install deprecated C libraries @<:@yes@:>@])]
```
Comment (wenningerk) [inline configure.ac:244]: Build/Install ... maybe as a hint that if not enabled there is nothing left that would check the code in any way

**Comment by wenningerk** [comment]
looks good

**Comment by kgaillot** [comment]
updated commits per review

**Pull request merged**